quantum/wpm.c

```diff
@@ -45,46 +34,105 @@ __attribute__((weak)) bool wpm_keycode_user(uint16_t keycode) {
     return false;
 }
 
-#ifdef WPM_ALLOW_COUNT_REGRESSION
-__attribute__((weak)) uint8_t wpm_regress_count(uint16_t keycode) {
-    bool weak_modded = (keycode >= QK_LCTL && keycode < QK_LSFT) || (keycode >= QK_RCTL && keycode < QK_RSFT);
+/* The shorter you make the capture duration the quicker the fall off once you stop typing. */
+/* Comment this out to get a longer fall-off but also a longer ramp up time to display */
+/* the actual WPM */
+#define WPM_SHORT_WINDOW
 
-    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) || (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX) || (keycode >= QK_MODS && keycode <= QK_MODS_MAX)) {
-        keycode = keycode & 0xFF;
-    } else if (keycode > 0xFF) {
-        keycode = 0;
-    }
-    if (keycode == KC_DEL || keycode == KC_BSPC) {
-        if (((get_mods() | get_oneshot_mods()) & MOD_MASK_CTRL) || weak_modded) {
-            return WPM_ESTIMATED_WORD_SIZE;
-        } else {
-            return 1;
+#ifndef WPM_SHORT_WINDOW
```

Comment:
This isn't the proper way to handle this... 

Either: 
```suggestion
#ifdef WPM_SHORT_WINDOW
```
or
```suggestion
#ifdef WPM_LONG_WINDOW
```

And in either case, this would need to be documented, as well.  However, the code does look to function better, overall.